src/caches/sdirk_caches.jl

```diff
@@ -176,6 +176,64 @@ function alg_cache(alg::SDIRK2,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUn
   SDIRK2Cache(u,uprev,fsalfirst,z₁,z₂,atmp,nlsolver)
 end
 
+struct SDIRK22ConstantCache{U,R,NR,Tab} <: OrdinaryDiffEqConstantCache
+  uhold::U
+  rhs::R
+  nl_rhs::NR
+  tab::Tab
+end
+
+function alg_cache(alg::SDIRK22,u,rate_prototype,uEltypeNoUnits,tTypeNoUnits,uBottomEltypeNoUnits,
+                   uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{false}})
+  # 1-1/√2
+  a = 0.29289321881345254
+  # -√2
+  α = -1.4142135623730951
+  # 1+√2
+  β = 2.414213562373095
+  tab = (a,α,β)
+
+  uhold = Vector{typeof(u)}(undef, 1)
+  rhs = ImplicitRHS_Scalar(f,zero(u),t,t,t,p)
+  nl_rhs = alg.nlsolve(Val{:init},rhs,uhold)
+  SDIRK22ConstantCache{typeof(uhold),typeof(rhs),typeof(nl_rhs),typeof(tab)}(uhold,rhs,nl_rhs,tab)
+end
+
+@cache mutable struct SDIRK22Cache{uType,DiffCacheType,uNoUnitsType,rateType,rhsType,nl_rhsType,Tab} <: OrdinaryDiffEqMutableCache
+  u::uType
+  uprev::uType
+  uprev2::uType
+  dual_cache::DiffCacheType
+  tmp::uType
+  atmp::uNoUnitsType
+  k::rateType
+  fsalfirst::rateType
+  rhs::rhsType
+  nl_rhs::nl_rhsType
+  tab::Tab
+end
+
+function alg_cache(alg::SDIRK22,u,rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,uprev2,f,t,dt,reltol,p,calck,::Type{Val{true}})
+   # 1-1/√2
+  a = 0.29289321881345254
+  # -√2
+  α = -1.4142135623730951
+  # 1+√2
+  β = 2.414213562373095
+  tab = (a,α,β)
+
+  tmp = similar(u); atmp = similar(u,uEltypeNoUnits)
+  k = zero(rate_prototype)
+  dual_cache = DiffCache(u,Val{determine_chunksize(u,get_chunksize(alg.nlsolve))})
+  rhs = ImplicitRHS(f,tmp,t,t,t,dual_cache,p)
```

Comment:
I think you looked at the wrong thing for inspiration. Take a look at Trapezoid, not GenericXXX

Reply:
I just had a look at how the newton method is implemented and it makes sense now. I will follow the Trapezoid method and update the code when I get some time. Thanks for the help!